services/api/src/models/group.ts

```diff
@@ -202,28 +219,30 @@ export const Group = (clients: {
         return groups.concat(flatSubGroups);
       };
 
-      groupId = R.pipe(
+      group = R.pipe(
         R.reduce(flattenGroups, []),
-        R.filter(R.propEq('name', name)),
-        R.path(['0', 'id'])
-      )(keycloakGroups) as string;
+        R.filter(R.propEq('name', name))
+      )(keycloakGroups);
 
-      if (R.isNil(groupId)) {
+      if (R.isNil(group)) {
```

Comment:
```suggestion
      if (R.isEmpty(group)) {
```
`group` will always be at least an empty array `[]` which means `R.isNil` is always false